lib/dal/src/workspace_snapshot/node_weight/component_node_weight.rs

```diff
@@ -212,28 +211,42 @@ impl CorrectTransforms for ComponentNodeWeight {
                     && is_self(source)
                     && destination.node_weight_kind == NodeWeightDiscriminants::SchemaVariant =>
                 {
-                    // Root props and sockets get all new AttributeValues during upgrade, but
-                    // the RemoveEdges for the old ones may be stale; RemoveEdge the real ones
-                    // just in case.
-                    remove_edges
-                        .extend(graph.outgoing_edges(component_node_idx, EdgeWeightKind::Root));
-                    schema_variant_changed = true;
-                    remove_edges.extend(
-                        graph.outgoing_edges(component_node_idx, EdgeWeightKind::SocketValue),
-                    );
+                    schema_variant_new_edge = true;
+                }
+                Update::RemoveEdge {
+                    source,
+                    destination,
+                    edge_kind,
+                } if EdgeWeightKindDiscriminants::Use == *edge_kind
+                    && is_self(source)
+                    && destination.node_weight_kind == NodeWeightDiscriminants::SchemaVariant =>
+                {
+                    schema_variant_remove_edge = true;
                 }
 
                 _ => {}
             }
         }
 
+        // For us to know that a schema variant upgrade is occurring, we need to confirm that there
+        // is both a new edge and a remove edge update from the component to a schema variant node.
+        let is_schema_variant_upgrade = schema_variant_new_edge && schema_variant_remove_edge;
+        if is_schema_variant_upgrade {
+            // Root props and sockets get all new AttributeValues during upgrade, but
+            // the RemoveEdges for the old ones may be stale; RemoveEdge the real ones
+            // just in case.
+            remove_edges.extend(graph.outgoing_edges(component_node_idx, EdgeWeightKind::Root));
+            remove_edges
+                .extend(graph.outgoing_edges(component_node_idx, EdgeWeightKind::SocketValue));
+        }
+
         if component_will_be_deleted {
             // Also remove any incoming edges to the component in case there
             // is a frame contains in another change set, or management edges
             remove_edges.extend(graph.edges_directed(component_node_idx, Incoming));
         }
 
-        if component_will_be_deleted || schema_variant_changed {
+        if component_will_be_deleted || is_schema_variant_upgrade {
```

Comment:
I believe this is the critical change to the control flow which no longer results in the error we saw?

Reply:
This particular one is updating code to account for the variable rename. The critical check is the `if is_schema_variant_upgrade` above this one.